internal/powerplatform/services/authorization/api_user.go

```diff
@@ -129,7 +129,11 @@ func (client *UserClient) CreateUser(ctx context.Context, environmentId, aadObje
 			break
 		}
 		tflog.Debug(ctx, fmt.Sprintf("Error creating user: %s", err.Error()))
-		client.Api.Sleep(10 * time.Second)
+		err = client.Api.SleepWithContext(ctx, 10*time.Second)
```

Comment:
All these hard-coded time durations would probably be better managed in a variable somewhere.

Reply:
Agree ... will create a separate issue / PR to track this since it's outside the scope of the billing policy bug I was trying to fix.